verification/src/uncles_verifier.rs

```diff
@@ -93,6 +95,12 @@ where
                 return Err(Error::Uncles(UnclesError::InvalidNumber));
             }
 
+            if !(included.contains(uncle.header.parent_hash())
+                || self.provider.descendant(uncle.header.parent_hash()))
+            {
+                return Err(Error::Uncles(UnclesError::DescendantLimit));
+            }
```

Comment:
Should check the uncle parent's number is just one less than the uncle's number. Either in descendant or uncle verifier. Otherwise, the uncle's parent could be any history block with a not matched continuous number.

Reply:
no it's not our rule